internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -240,41 +218,52 @@ func WithConnectionPublisher(p ConnectionPublisher) ReconcilerOption {
 type compositeResource struct {
 	CompositionSelector
 	Configurator
+	Composer
```

Comment:
I like that we separated `Composer`. We might be able to use the same reconciler in `CustomComposition` by changing the `Composer` implementation.

Reply:
That's good to know - I was worried it might be going back to too many layers of abstraction but it seemed like a clean-ish way to have two different implementations.